CLUSTER-SETUP.md

```diff
@@ -0,0 +1,92 @@
+
+## Setup: Kubernetes Cluster
+You will need a Kubernetes Cluster (either single or multi-node) with `kubectl` and `helm` (v3) to talk to the cluster.
+
+### Running locally (developer only)
+Several options exist for running a Kubernetes cluster locally:
+* Kubernetes under Docker for MacOSX/Windows - you can enable Kubernetes in the Settings and install the `helm` client
+* `minikube` - you will need to set up a custom domain pointing to your `minikube ip`
+
+### Remote VM Single-node
+For a short 3-step process for setting this up on a remote VM (where `minikube` may be unavailable), check out [our fork of Data8's kubeadm-bootstrap](https://github.com/nds-org/kubeadm-bootstrap)
+This will install Kubernetes via `kubeadm` and configure it to run as a single node cluster. It will also deploy the NGINX Ingress controller to the cluster, allowing you to skip the steps for deploying it manually (provided below).
+
+#### Manually Scaling Up Additional Nodes
+Once the script has finished running, it should output instructions for how to join more nodes to this cluster.
+
+To manually add a node to the cluster, copy and paste the `kubeadm join ..... ` command from your first node's console to another kubeadm-enabled VM and kubeadm will set up that VM as a worker node in your cluster. 
+
+### Remote VM Multi-node via Terraform
+For a more robust Workbench cluster the spans multiple OpenStack VMs, you can use our [kubeadm-terraform](https://github.com/nds-org/kubeadm-terraform) plan to spin up a cluster of your desired size and scale.
+
+## StorageClass / Volume Provisioner
+At least one StorageClass needs to be configured in your cluster as the default.
+GKE and AWS EKS will provide these by default, but OpenStack does not offer the same out of the box.
+If you have already chosen and configured a StorageClass that can privision PersistentVolume (PVs) for you, then you can skip this step.
+
+To check the StorageClasses in your cluster:
+```bash
+$ kubectl get storageclass
+NAME                 PROVISIONER          RECLAIMPOLICY   VOLUMEBINDINGMODE   ALLOWVOLUMEEXPANSION   AGE
+hostpath (default)   docker.io/hostpath   Delete          Immediate           false                  32d
+```
+
+If you don't see any listed with `(default)`, you'll need to set one as the default by following [these steps](https://kubernetes.io/docs/tasks/administer-cluster/change-default-storage-class/).
+
+If you do not see *any* StorageClass listed, then you'll need to set one up first (see below).
+
+### Setup: NFS Server Provisioner
+
+While there are many volume provisioners available, we tend to use the [NFS Server Provisioner](https://github.com/nds-org/kubeadm-terraform/tree/develop/assets/nfs) to provision volumes for Workbench.
+
+To use the NFS Server Provisioner in your cluster, run the following command:
+```bash
+$ kubectl apply -f https://github.com/nds-org/kubeadm-terraform/tree/develop/assets/nfs
+```
+
```

Comment:
When I try to run this, the following error was shown up. 

> error: error parsing https://github.com/nds-org/kubeadm-terraform/tree/develop/assets/nfs: error converting YAML to JSON: yaml: line 138: mapping values are not allowed in this context

However, if I applied it after cloning, it's working well. What's the difference between them?

Reply:
Thank you, good catch :+1: this was a bad URL and I clearly need to test these instructions more carefully